crt0stack/examples/example.rs

```diff
@@ -0,0 +1,56 @@
+use crt0stack::{AuxvEntry, Crt0Stack};
+use std::string::ToString;
+
+fn main() {
+    const ELF64_HDR_SIZE: u64 = 0x40;
+    const ELF64_PHDR_SIZE: u64 = 56;
+    let load_addr: u64 = 0x40_0000; // First elf PT_LOAD
+    let num_elf_program_sections: usize = 4;
+
+    let mut crt0sp = Crt0Stack::new();
+    crt0sp.envp.push("LANG=C".to_string());
+    crt0sp.argv.push("/init".to_string());
+    crt0sp.add_auxv_entry(AuxvEntry::ExecFilename("/init".to_string()));
```

Comment:
The code block initializing the `Crt0Stack` looks like it would be a good fit for the builder pattern.

Reply:
Well, Crt0Stack is the builder filling in a &[u8]

Reply:
Ok, implemented the owning builder pattern